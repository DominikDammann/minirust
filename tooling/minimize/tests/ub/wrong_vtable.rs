trait A {
    #[allow(dead_code)]
    fn foo(&self) -> usize;
}

trait B {}

impl A for usize {
    fn foo(&self) -> usize {
        *self
    }
}

impl B for usize {}

fn main() {
    let a: &dyn A = &42_usize;
    let _b = unsafe { core::mem::transmute::<&dyn A, &dyn B>(a) };
}
