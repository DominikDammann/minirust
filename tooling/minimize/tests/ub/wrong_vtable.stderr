fatal error: UB: Value::Ptr: invalid vtable in metadata
